addons/controllers/clusterbootstrap_controller.go

```diff
@@ -132,11 +133,14 @@ func (r *ClusterBootstrapReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, err
 	}
 
-	tkrName := util.GetTKRNameForCluster(r.context, r.Client, cluster)
-	if tkrName == "" {
-		log.Info("cluster does not have an associated TKR")
+	// make sure the TKR object exists
+	tkrName := cluster.Labels[constants.TKRLabelClassyClusters]
+	tkr, err := util.GetTKRByName(r.context, r.Client, tkrName)
+	if err != nil || tkr == nil {
+		log.Error(err, "unable to fetch TKR object", "name", tkrName)
 		return ctrl.Result{}, nil
```

Comment:
We don't want to swallow `err`.
```suggestion
		return ctrl.Result{}, err
```

The only case when we want that, (looks like it) should be handled by `util.GetTKRByName()` and it's when `apierrors.IsNotFound(err) == true`